Add ES modules, classes.

mbostock opened the pull request:
I modeled this after our earlier work on Delaunator, but I’m happy to make any changes if you’d prefer a different approach.

mourner commented on package.json:
```diff
@@ -26,9 +38,19 @@
+    "esm": "^3.0.8",
+    "rollup": "^0.57.1",
+    "rollup-plugin-buble": "^0.19.2",
+    "rollup-plugin-uglify": "^3.0.0",
```
Also let's use newer versions here.

mbostock commented:
I’ve updated the Node version for Travis to 6 as requested, and updated Rollup to 0.64. It looks like rollup-plugin-uglify no longer supports transpiling ES6+ syntax, and while we don’t currently need that (since we aren’t publishing a minified ES module), I figured we might as well switch to rollup-plugin-terser.

mourner commented:
Thanks! Published as v2.0.0.
